src/panoramic/cli/validate.py

```diff
@@ -64,6 +72,33 @@ def _validate_file(fp: Path, schema: Dict[str, Any]):
         raise JsonSchemaError(path=fp, error=e)
 
 
+def _validate_fields(fields: Iterable[Tuple[Dict[str, Any], Path]]) -> List[ValidationError]:
+    field_paths_by_id: Dict[Tuple, List[Path]] = defaultdict(list)
+    errors = []
+    for field_data, field_path in fields:
+        try:
+            _validate_data(field_data, JsonSchemas.field())
+            field = PanoField.from_dict(field_data)
+            field_paths_by_id[field.id].append(field_path)
```

Comment:
It's kind of non obvious what's going on here with the company/dataset scoping... can we either change `id` to be `Tuple[Optional[str], str]` or even better, use `field_paths_by_id[(field.data_source, field.slug)] = ...` here?

That way, you can replace the iteration below with
```
for (data_source, field_slug), paths in field_paths_by_id.items():
```
?

Reply:
Done c0ffb48.